### PR TITLE
Sim translator node

### PR DESCRIPTION
**Pull request by alistair-english**

### CODE REVIEW

File: src/navigation/sim_translator/package.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+    <name>sim_translator</name>
+    <version>0.0.0</version>
+    <description>Translates FS Sim topics to Driverless topics</description>
+    <maintainer email="team@qutmotorsport.com">Alistair English, Alastair Bradford</maintainer>
+    <license>MIT</license>
+
```
Comment (b1n-ch1kn) [inline src/navigation/sim_translator/package.xml:9]: would add `<depend>` for `driverless_msgs` + `driverless_common` so we can build `--packages-up-to` and will include that

File: src/navigation/sim_translator/sim_translator/node_cone_detection_translator.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+class ConeDetectonTranslator(Node):
+    track: List[FSCone] = []
+
+    def __init__(self) -> None:
+        super().__init__("cone_detection_translator")
+        self.create_subscription(Track, "/testing_only/track", self.track_callback, 10)
+        self.create_subscription(Odometry, "/testing_only/odom", self.odom_callback, 10)
+
+        self.detection_publisher: Publisher = self.create_publisher(ConeDetectionStamped, "cone_detection", 1)
```
Comment (b1n-ch1kn) [inline src/navigation/sim_translator/sim_translator/node_cone_detection_translator.py:28]: we've got a few other `cone_detection` topics now, maybe `/translator/cone_detection` to ease confusion?

File: src/navigation/sim_translator/sim_translator/node_cone_detection_translator.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        x = odom_msg.pose.pose.position.x
+        y = odom_msg.pose.pose.position.y
+
+        detected_cones: List[Cone] = []
+        for cone in self.track:
+            # displacement from car to cone
+            x_dist = cone.location.x - x
+            y_dist = cone.location.y - y
+            range_ = sqrt(x_dist**2 + y_dist**2)
```
Comment (b1n-ch1kn) [inline src/navigation/sim_translator/sim_translator/node_cone_detection_translator.py:53]: i dislike this

File: src/common/driverless_common/driverless_common/marker.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        r, g, b = 255, 255, 0
+    elif cone_colour in [Cone.ORANGE_BIG, Cone.ORANGE_SMALL]:
+        r, g, b = 255, 0, 0
+    else:
+        r, g, b = 0, 0, 0
+
+    marker.color.r = float(r)
+    marker.color.g = float(g)
+    marker.color.b = float(b)
+    marker.color.a = 1.0
```
Comment (b1n-ch1kn) [inline src/common/driverless_common/driverless_common/marker.py:80]: was looking into this more, stuff like position, orientation, colour can be done in 1 line really. not a problem, just means we dont have a 50 line message translator

**Review by b1n-ch1kn — CHANGES_REQUESTED** [review]

File: src/common/driverless_common/driverless_common/marker.py (hunk 4 above)
Comment (alistair-english) [reply src/common/driverless_common/driverless_common/marker.py:80]: yeah i like that

**Review by b1n-ch1kn — APPROVED** [review]

**Pull request merged**